In [2]:
import pandas as pd
from statsmodels.tsa.stattools import adfuller, kpss

# 读取数据
data = pd.read_csv('data.csv',encoding='gbk')

# 提取航线金额列作为时间序列数据
time_series = data['航线金额']

# ADF检验
adf_result = adfuller(time_series)
adf_pvalue = adf_result[1]
print("ADF p-value:", adf_pvalue)

# KPSS检验
kpss_result = kpss(time_series)
kpss_pvalue = kpss_result[1]
print("KPSS p-value:", kpss_pvalue)

ADF p-value: 0.2097850995366165
KPSS p-value: 0.01


C:\Users\510\AppData\Local\Temp\ipykernel_31200\3797843963.py:16: InterpolationWarning: The test statistic is outside of the range of p-values available in the
look-up table. The actual p-value is smaller than the p-value returned.

  kpss_result = kpss(time_series)


数据没有平稳性！

- ADF检验：根据ADF检验的p值（adf_pvalue），如果p值小于显著性水平（通常为0.05），则可以拒绝原假设（数据具有单位根，即非平稳性），并认为数据是平稳的。在这种情况下，p值为0.2098大于0.05，因此无法拒绝原假设，不能确定数据是平稳的。

- KPSS检验：根据KPSS检验的p值（kpss_pvalue），如果p值小于显著性水平（通常为0.05），则可以拒绝原假设（数据是平稳的），并认为数据具有非平稳性。在这种情况下，p值为0.01小于0.05，因此可以拒绝原假设，认为数据具有非平稳性。